.golangci.yml

```diff
@@ -15,11 +15,15 @@ issues:
   exclude-use-default: false
 
   exclude-rules:
-   # Exclude duplicate code checking in test files (due to common repeats in
-   # test setup code)
-   - path: _test\.go
+   # Exclude duplicate code and function length and complexity checking in test
+   # files (due to common repeats and long functions in test code)
+   - path: _test\.go #TODO: also exclude all go-generated files by giving them common names? *_gen.go?
      linters:
+       - gocyclo
        - dupl
+       - gocognit
```

Comment:
I'm not sure if we should disable `gocyclo` and `gocognit` globally for all tests, since it could point out some hairy looking tests, though we'll probably notice those in a review.

How about disabling these on a per-test basis?

Reply:
My thinking here is that the complexity threshold for tests us usually significantly higher, since most of it is just boilerplate setup and verification code. So I personally would prefer not to deal with manually disabling these warnings everywhere, and as you say, if some test is egregiously complex, just to handle that in the code review process.